CMakeLists.txt

```diff
@@ -1,32 +1,34 @@
 cmake_minimum_required(VERSION 3.3)
 cmake_policy(VERSION 3.3)
 
-# get version from Python package
-find_package(PythonInterp REQUIRED)
-execute_process(COMMAND ${PYTHON_EXECUTABLE} setup.py --version
-                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-                RESULT_VARIABLE PY_VERSION_FAILED
-                OUTPUT_VARIABLE PY_PKG_VERSION_FULL
-                ERROR_VARIABLE PY_VERSION_ERROR
-                OUTPUT_STRIP_TRAILING_WHITESPACE)
-if(PY_VERSION_FAILED)
-  message(
-    FATAL_ERROR
-      "Could not retrieve version from command '${PYTHON_EXECUTABLE} setup.py --version'\n"
-      ${PY_VERSION_ERROR})
-endif()
+if(NOT SONATA_VERSION)
+    execute_process(COMMAND git describe --tags
+                    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+                    RESULT_VARIABLE GIT_VERSION_FAILED
+                    OUTPUT_VARIABLE GIT_PKG_VERSION_FULL
+                    ERROR_VARIABLE GIT_VERSION_ERROR
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(GIT_VERSION_FAILED)
+      message(
+        FATAL_ERROR
+          "Could not retrieve version from command 'git describe --tags'\n"
+          ${GIT_VERSION_ERROR})
+    endif()
+
 # keep last line of command output
-string(REPLACE "\n"
-               ";"
-               PY_PKG_VERSION_FULL
-               "${PY_PKG_VERSION_FULL}")
-list(GET PY_PKG_VERSION_FULL -1 PY_PKG_VERSION_FULL)
+    string(REPLACE "\n"
+                   ";"
+                   GIT_PKG_VERSION_FULL
+                   "${GIT_PKG_VERSION_FULL}")
+   list(GET GIT_PKG_VERSION_FULL -1 SONATA_VERSION)
+endif()
+
 # keep MAJOR.MINOR.PATCH (PATCH being optional)
```

Comment:
What do people think about making 'patch' not be optional, and setting the CMake's `<tweak>` to be the count of commits from the since the tag?